### PR TITLE
[SPARK-45221][PYTHON][DOCS] Refine docstring of DataFrameReader.parquet

### PR DESCRIPTION
**Pull request by allisonwang-db**
### What changes were proposed in this pull request?

This PR refines the docstring of DataFrameReader.parquet by adding more examples.

### Why are the changes needed?

To improve PySpark documentation

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

doctest

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by allisonwang-db** [comment]
The test failure is unrelated. cc @zhengruifeng @HyukjinKwon

File: python/pyspark/sql/readwriter.py [hunk 1]
```diff
@@ -505,24 +506,69 @@ def parquet(self, *paths: str, **options: "OptionalPrimitiveType") -> "DataFrame
...
+
+        Read multiple Parquet files and merge schema.
+
+        >>> with tempfile.TemporaryDirectory() as d1, tempfile.TemporaryDirectory() as d2:
+        ...     df.write.mode("overwrite").format("parquet").save(d1)
+        ...     df2.write.mode("overwrite").format("parquet").save(d2)
+        ...
+        ...     spark.read.option("mergeSchema", "true") \\
+        ...         .parquet(d1, d2).select("name", "age", "height") \\
+        ...         .orderBy("name", "age").show()
```
Comment (HyukjinKwon) [inline python/pyspark/sql/readwriter.py:562]:
```suggestion
        ...     spark.read.option(
        ...         "mergeSchema", "true"
        ...     ).parquet(d1, d2).select(
        ...         "name", "age", "height"
        ...     ).orderBy("name", "age").show()
```

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/readwriter.py [hunk 2]
```diff
@@ -505,24 +506,71 @@ def parquet(self, *paths: str, **options: "OptionalPrimitiveType") -> "DataFrame
...
+        ...     ).orderBy("name", "age").show()
+        +-----+----+------+
+        | name| age|height|
+        +-----+----+------+
+        |Alice|null|    70|
+        |Alice|  10|  null|
+        |  Bob|null|    80|
+        |  Bob|  15|  null|
+        |  Tom|  20|  null|
+        +-----+----+------+
```
Comment (HyukjinKwon) [inline python/pyspark/sql/readwriter.py:573]:
```suggestion
        +-----+----+------+
        | name| age|height|
        +-----+----+------+
        |Alice|NULL|    70|
        |Alice|  10|  NULL|
        |  Bob|NULL|    80|
        |  Bob|  15|  NULL|
        |  Tom|  20|  NULL|
        +-----+----+------+
```

File: python/pyspark/sql/readwriter.py [hunk 3]
```diff
@@ -505,24 +506,71 @@ def parquet(self, *paths: str, **options: "OptionalPrimitiveType") -> "DataFrame
...
+        |age| name|
+        +---+-----+
+        | 10|Alice|
+        | 15|  Bob|
+        | 20|  Tom|
+        +---+-----+
+
+        Read a Parquet file with a specific column.
+
+        >>> with tempfile.TemporaryDirectory as d:
```
Comment (HyukjinKwon) [inline python/pyspark/sql/readwriter.py:541]:
```suggestion
        >>> with tempfile.TemporaryDirectory() as d:
```

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by allisonwang-db** [comment]
Thanks @HyukjinKwon!

**Pull request closed without merging**